patches/seashell.py

```diff
@@ -62,3 +62,45 @@ def upgradeMansion(rom):
         ld   a, [$DAE9]
         and  $10
     """))
+
+    # Do not reset seashell counter
+    rom.patch(0x19, 0x3700, ASM("ld [wSeashellsCount], a"), "", fill_nop=True)
+    # Prevent missing out on a reward or getting the last reward multiple times
+    rom.patch(0x19, 0x31CA, ASM("ld a, [wSeashellsCount]"), ASM("call $7F30"))
+    rom.patch(0x19, 0x3215, ASM("ld a, [wSeashellsCount]"), ASM("call $7F30"))
+    rom.patch(0x19, 0x32A2, ASM("ld a, [wSeashellsCount]"), ASM("call $7F30"))
+    rom.patch(0x19, 0x38B3, ASM("ld a, [wSeashellsCount]"), ASM("call $7F30"))
+    # This is using free space in bank 0x19 because we need to return a value in A when called
+    rom.patch(0x19, 0x3F30, 0x3F60, ASM("""

```

Comment:
When using free space, it's better to use `"00" * [size]` as input then an fixed end address.
See: https://github.com/daid/LADXR/blob/master/patches/enemies.py#L596

This ensures that no two patches are using the same "free space" by mistake.

(I still need to fully review these ASM patches, it's always complex to read any ASM patch, so it takes some time and the right mindset)

Reply:
Good suggestion, I'll change that.
Yeah, I get that about reading ASM, take your time.
I'm using the B register at one point as a parameter that persists through the bank switch. It should hopefully be clear when comparing to the disassembly.